src/DotNetEnv/Parsers.cs

```diff
@@ -270,9 +270,10 @@ private static string ToUtf32Char (string hex)
 
         public static IEnumerable<KeyValuePair<string, string>> ParseDotenvFile (
             string contents,
-            Func<KeyValuePair<string, string>, KeyValuePair<string, string>> tranform
+            Func<KeyValuePair<string, string>, KeyValuePair<string, string>> transform,
+            bool enableInterpolation = true
         ) {
-            return Assignment.Select(tranform).Or(Empty).AtLeastOnce().End()
+            return Assignment(enableInterpolation).Select(transform).Or(Empty).AtLeastOnce().End()
```

Comment:
Thank you for correcting the typo.

I think your approach is possibly fine, if this change is indeed desirable, but I would consider passing the object to use for values. Currently it is `Value`, but you could have an alternative called Raw value, so that you did not need the logic inside of Value, and most notably the copy of both interpolated and raw values, but just the raw value every time.

Reply:
Quite some time ago since I made these changes ;)

I'm not really sure what you exactly mean. Somehow I need to "inject" whether we need to Interpolate or not.

What I understand is, that the Parser (somewhere down the InterpolatedParser) should return a "RawValue" (which would be the same as Value I guess) instead of an "InterpolatedValue".
The problem I see (without looking into the code right now) is, that we have to propagate the Parameter `enableInterpolation` all the way down to the InterpolatedParser.

Not sure if I understand it correctly, maybe you can give me additional advice.
I will have another look as soon as I find some more time for it.

Reply:
Checked it again now.
Maybe your intention was to do something like that?
```
from value in interpolationEnabled ? Value : RawValue
```

I guess that would be possible, but it would introduce a whole new Parsing-Tree, which should behave exactly like Value, except for the Interpolation part of course.
Not sure how to achieve that without making this whole thing really complicated. I guess it would end in passing parameters too, to be able to use the same Parsers. And that would mean to pass the `interpolationEnabled` Parameter down into the `InterpolatedValue`-Parsers and return a not Interpolated value from there.
And that's more or less the same than just passing the interpolationEnabled down the whole road without a differently named Parser.

I'm not really sure if that is something we should aim for, sounds harder to understand for me.
Is that the direction you wanted to point me or did I misunderstood something?

Reply:
One more word about it:
Long-time goal imho would be to not interpolate "inside" the Parser at all. Or at least pushing the Interpolation completely after the Parsing-Operation.

That would give the possibility to have Parser-Results, which are as near to the original EnvFile as possible. And further interpretation (like Interpolation, clobbering, setEnvVars etc.) could be done completely independent, which would allow more independant tests for this additional functionality, which is not really dependant to the Parsing and vice versa. In my opinion these are different aspects, which should be handled independently too.